src/app/main/component/auth/components/fragments/input-fields/username-input-field/username-input-field.component.ts

```diff
@@ -13,25 +17,41 @@ import { AbstractControl, FormControl, FormGroup, NG_VALUE_ACCESSOR, Validators
     }
   ]
 })
-export class UsernameInputFieldComponent implements OnInit {
-  public passwordErrorMessageBackEnd: string;
+export class UsernameInputFieldComponent implements OnInit, OnDestroy {
+  private destroy: Subject<boolean> = new Subject<boolean>();
   public backEndError: string;
   public firstNameErrorMessageBackEnd: string;
   public userNameInputForm: FormGroup;
   public nameField: AbstractControl;
   public userNameValue: string;
 
-  constructor() {}
+  constructor(private popUpViewService: PopUpViewService) {}
 
   ngOnInit(): void {
+    this.popUpViewService.firstNameBackendErrorSubject
+      .pipe(takeUntil(this.destroy))
+      .subscribe((value) => (this.firstNameErrorMessageBackEnd = value));
     this.userNameInputForm = new FormGroup({
-      firstName: new FormControl(null)
+      firstName: new FormControl('', [Validators.required])
     });
+    this.userNameInputForm.setValidators(ValidatorRegExp('firstName'));
     this.nameField = this.userNameInputForm.get('firstName');
   }
 
   public setError(): void {
-    this.passwordErrorMessageBackEnd = null;
+    this.firstNameErrorMessageBackEnd = null;
+    this.backEndError = null;
     this.userNameValue = this.nameField.value;
+    this.nameField.markAsTouched();
+    if (this.nameField.valid && this.userNameInputForm.touched) {
+      this.popUpViewService.setUserNameInputField(true);
+    } else {
+      this.popUpViewService.setUserNameInputField(false);
+    }
```

Comment:
remove if...else
const isValue = this.nameField.valid && this.userNameInputForm.touched
this.popUpViewService.setUserNameInputField(isValue);

Reply:
done